daemon/daemon.go

```diff
@@ -619,6 +622,9 @@ func (d *Daemon) rebootDelay() (time.Duration, error) {
 		return 0, err
 	}
 	rebootDelay := 1 * time.Minute
+	if immediate {
+		rebootDelay = 0
+	}
```

Comment:
this looks ok, but means SNAPD_REBOOT_DELAY gets applied for "immediate" as well.  prepare.sh has:

Environment=... SNAPD_REBOOT_DELAY=10m ...

this probably doesn't affect install but might affect other tests later? should we do something about that?